library/Vanilla/Formatting/Formats/RichFormat.php

```diff
@@ -271,4 +271,53 @@ private function logBadInput(string $input) {
             E_USER_WARNING
         );
     }
+
+    /**
+     * Filter a rich body to remove sensitive information.
+     *
+     * @param array $row The row to filter.
+     * @return array Returns the filtered row.
+     */
+    public static function editBodyFilter($row) {
+        if (!is_array($row) || strcasecmp($row['format'] ?? $row['Format'] ?? '', 'rich') !== 0) {
+            return $row;
+        }
+
+        $key = array_key_exists('Body', $row) ? 'Body' : 'body';
+        $row[$key] = self::stripSensitiveInfoRich($row[$key]);
+
+        return $row;
+    }
+
+    /**
+     * Strip sensitive user info from a rich string and rewrite it.
+     *
+     * @param string $input The rich text input.
+     * @return string The string.
+     */
+    private static function stripSensitiveInfoRich(string $input): string {
+        if (strpos($input, "password") === false) {
```

Comment:
Should this be `stripos` ?

Reply:
In this case I think it's okay because APIv2 camel cases everything.